realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -859,21 +861,270 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableQuery_nativeFind(
     return -1;
 }
 
+JNIEXPORT jlong JNICALL Java_io_realm_internal_TableQuery_nativeFindWithHandover(
```

Comment:
This function returns a native pointer as a Java `long`. It seems safer to return 0 instead of -1 in cases of errors. From inspecting native pointers in Java, I have noticed that they are often returned as negative values, and -1 might be interpreted as a valid pointer.